library/objective-c/EnvoyEngine.h

```diff
@@ -312,11 +312,15 @@ extern const int kEnvoyFailure;
  @param logLevel The log level to use when starting Envoy.
  @param onEngineRunning Closure called when the engine finishes its async startup and begins
  running.
+ @param onEngineLog Closure to be called when the engine logs.
+ @param onEngineFlush Closure to be called when the engine's logger flushes.
  @return A status indicating if the action was successful.
  */
 - (int)runWithConfig:(EnvoyConfiguration *)config
             logLevel:(NSString *)logLevel
-     onEngineRunning:(nullable void (^)())onEngineRunning;
+     onEngineRunning:(nullable void (^)())onEngineRunning
+         onEngineLog:(nullable void (^)(NSString *))onEngineLog
```

Comment:
nit: Let's call just call this a Logger

Reply:
Given that this is just a block I want to keep the name for consistency. However, I do share the sentiment you wrote in for both the bridge, and at this level too:

> Not I'm a little concerned about the increase in things appearing in the Engine constructor. Another option would be to have this be runtime configurable. We don't need it right now, but we should maybe also think about the case where people also might want to add more than one logger. [source](https://github.com/envoyproxy/envoy-mobile/pull/1355#issuecomment-814389318)

And would like better separation of concerns (engine callbacks, config, logging, pulse api) at this level. I can carve this out here to begin that process and not add more to the pile that already exists, or we can land as is, and then start a series of PRs to clean up.

Reply:
Everything's in the same header due to the objective-c bridge header clang issue - if that's what you mean by separation of concerns.